File: src/main/java/com/aditi/moneymanager/controller/NotesController.java
package com.aditi.moneymanager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.aditi.moneymanager.model.NotesModel;
import com.aditi.moneymanager.service.NotesService;

@RestController
public class NotesController {

    @Autowired
    private NotesService service;

    @GetMapping("/notes")
    public ResponseEntity<List<NotesModel>> getNotes(Authentication req) {
        List<NotesModel> notes = service.getNotes(req.getName());
        return ResponseEntity.ok(notes);
    }

    @PostMapping("/notes")
    public ResponseEntity<NotesModel> addNote(Authentication req, @RequestBody Map<String, String> payload) {
        NotesModel noteModel = service.addNote(req.getName(), payload);
        return ResponseEntity.ok(noteModel);
    }

    @PutMapping("/notes/{id}")
    public ResponseEntity<NotesModel> updateNote(Authentication req, 
            @PathVariable Long id, @RequestBody Map<String, String> payload) {
        NotesModel updatedNote = service.updateNote(req.getName(), id, payload.get("note"));
        return ResponseEntity.ok(updatedNote);
    }

    @DeleteMapping("/notes/{id}")
    public ResponseEntity<String> deleteNote(Authentication req, @PathVariable Long id) {
        service.deleteNote(req.getName(), id);
        return ResponseEntity.ok("Note deleted successfully.");
    }
}

File: controllers/NotesController.js
import express from "express";
import NotesService from "../service/NotesService.js";

const router = express.Router();

router.get("/", async (req, res) => {
  try {
    const notes = await NotesService.getNotes(req.query.username);
    res.status(200).json(notes);
  } catch (error) {
    res.status(400).json({ message: error.message });
  }
});

router.post("/", async (req, res) => {
  try {
    const note = await NotesService.addNote(req.query.username, req.body.note);
    res.status(200).json(note);
  } catch (error) {
    res.status(400).json({ message: error.message });
  }
});

router.put("/:id", async (req, res) => {
  try {
    const updatedNote = await NotesService.updateNote(req.query.username, req.params.id, req.body.note);
    res.status(200).json(updatedNote);
  } catch (error) {
    res.status(400).json({ message: error.message });
  }
});

router.delete("/:id", async (req, res) => {
  try {
    await NotesService.deleteNote(req.query.username, req.params.id);
    res.status(200).json({ message: "Note deleted successfully." });
  } catch (error) {
    res.status(400).json({ message: error.message });
  }
});

export default router;